{"type": "indicator", "spec_version": "2.1", "id": "indicator--9ee2ee96-4aab-415a-b39e-b4a60fe7a62b", "created_by_ref": "identity--748e6444-f073-4c50-b558-f49be8897a81", "created": "2009-10-22T16:30:00.000Z", "modified": "2009-12-19T06:57:00.000Z", "name": "CVE-2009-2940", "description": "The pygresql module 3.8.1 and 4.0 for Python does not properly support the PQescapeStringConn function, which might allow remote attackers to leverage escaping issues involving multibyte character encodings.", "indicator_types": ["compromised"], "pattern": "[((software:cpe='cpe:2.3:a:pygresql:pygresql:3.8.1:*:*:*:*:*:*:*' OR software:cpe='cpe:2.3:a:pygresql:pygresql:4.0:*:*:*:*:*:*:*') AND (software:cpe='cpe:2.3:a:python:python:*:*:*:*:*:*:*:*'))]", "pattern_type": "stix", "pattern_version": "2.1", "valid_from": "2009-10-22T16:30:00Z", "external_references": [{"source_name": "cve2stix", "description": "This object was created using cve2stix from the Signals Corps.", "url": "https://github.com/signalscorps/cve2stix"}], "object_marking_refs": ["marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"], "extensions": {"extension-definition--b463c449-d022-48b7-b464-3e9c7ec5cf16": {"extension_type": "property-extension", "all_cpe23uris": ["cpe:2.3:a:pygresql:pygresql:3.8.1:*:*:*:*:*:*:*", "cpe:2.3:a:pygresql:pygresql:4.0:*:*:*:*:*:*:*", "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"], "vulnerable_cpeid": ["cpe:2.3:a:pygresql:pygresql:3.8.1:*:*:*:*:*:*:*", "cpe:2.3:a:pygresql:pygresql:4.0:*:*:*:*:*:*:*"]}}}